web/src/components/upload/index.js

```diff
@@ -1,25 +1,32 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import moment from "moment-timezone";
 
 import { UploadDrop } from "./drop";
 import { UploadConfirm } from "./confirm";
 import { Container } from "../ui";
+import { getDevices } from "./drop.utils";
 
 export const Upload = () => {
   const [files, setFiles] = useState([]);
   const [phase, setPhase] = useState("drop"); //"drop" or "confirm" page
   const [dustrakStart, setDustrakStart] = useState(moment(""));
   const [dustrakEnd, setDustrakEnd] = useState(moment(""));
   const [dustrakSerial, setDustrakSerial] = useState("");
+  const [devices, setDevices] = useState([]);
 
-  /**
-   * Map of each dustrak serial number to its label
-   */
-  const dustrakSerialToLabel = {
-    8530091203: "Device A",
-    8530094612: "Device B",
-    8530100707: "Device C",
-  };
+  useEffect(() => {
+    (async () => {
+      try {
+        const deviceList = await getDevices();
+        setDevices(deviceList);
+      } catch(err) {
+        console.error(err);
+      }
+    })()
+  }, [dustrakSerial]);
+
+  const matchedDevice = devices.filter(device => device.serial === dustrakSerial);
```

Comment:
1. As we're interested in finding one exact match, we can use `.find`, instead of `.filter`. It's also more efficient, as it can stop once it finds a match- rather than needing to look at every element, like filter does.
2. The function can be moved into `useEffect` in order to set the state of the device at the point where it would change.
3. It is a pure function, which means we can move it to the utility functions. This allows us to a. separate the concerns of each file and b. easily apply a unit test to it